app/web_framework/api_version_handler.py

```diff
@@ -0,0 +1,62 @@
+import re
+
+
+class APIVersionHandler():
+    OLD_VERSIONED_URI_COMPONENT = 'v1'
+    API_VERSION_HEADER_KEY = 'Api-Version'
+
+    _versions = [
+        1,
+        2
+    ]
+
+    @classmethod
+    def resolve_version(cls, accept_header_value: str, uri: str) -> int:
+        """
+        Get the respective version of the API relative to the request header and URI.
+        :param accept_header_value: The value of the header which to search for version type (Content-Type/Accept).
+        :param uri: The URI from the request being checked.
+        """
+        if accept_header_value is None:
+            return cls._get_default(uri)
+
+        version = cls._get_default(uri)
+        try:
+            matches = re.match(r'.*(?:application/vnd.clusterrunner.v(\d+)\+json).*',
```

Comment:
Instead of `re.match` with a leading and trailing `.*`, you can just use `re.search`.
https://docs.python.org/3.4/library/re.html#search-vs-match

Also maybe use `re.IGNORECASE` instead of `re.I` for readability.